src/librustc_mir/transform/generator.rs

```diff
@@ -406,9 +418,9 @@ fn locals_live_across_suspend_points<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
             live_locals.union(&ignored.0);
 
             if !movable {
-                // For immovable generators we consider borrowed locals to always be live.
+                // For immovable generators we consider locals to always be live once borrowed.
                 // This effectively makes those locals use just the storage liveness.
```

Comment:
I'd appreciate a code example here, or somewhere =), explaining the meaning of these bits in a more intuitive way

Reply:
I added some more comments. Let me know if that is good enough.